cluster/cluster.go

```diff
@@ -17,20 +19,54 @@ var (
 
 type Cluster struct {
 	sync.RWMutex
+	store           *state.Store
 	tlsConfig       *tls.Config
 	eventHandlers   []EventHandler
 	nodes           map[string]*Node
 	overcommitRatio float64
 }
 
-func NewCluster(tlsConfig *tls.Config, overcommitRatio float64) *Cluster {
+func NewCluster(store *state.Store, tlsConfig *tls.Config, overcommitRatio float64) *Cluster {
 	return &Cluster{
 		tlsConfig:       tlsConfig,
 		nodes:           make(map[string]*Node),
+		store:           store,
 		overcommitRatio: overcommitRatio,
 	}
 }
 
+// Deploy a container into a `specific` node on the cluster.
+func (c *Cluster) DeployContainer(node *Node, config *dockerclient.ContainerConfig, name string) (*Container, error) {
+	container, err := node.Create(config, name, true)
+	if err != nil {
+		return nil, err
+	}
+
+	// Commit the requested state.
+	st := &state.RequestedState{
```

Comment:
If we add fields in `RequestedState` and load an old version, there will be no error, the new fields will be empty.

If we remove fields in `RequestedState` and load an old version, there will be no error either.

The only error could be if we change a string to a float or something like this.

Does this answer your question ?